[react-instantsearch-nextjs@1.1.2] `Cannot find module 'next/headers'` at runtime on Vercel/Lambda after upgrade from 1.0.7

### 🐛 Current behavior

After upgrading `react-instantsearch-nextjs` from `1.0.7` to `1.1.2`, pages using `<InstantSearchNext>` fail at runtime on Vercel (serverless/Lambda) with:

```
Error: Cannot find module 'next/headers'
```


The build succeeds without errors, but all page requests served by Lambda return a 500.


> [!NOTE]
> The following problem analysis was done using AI Agents and could be wrong - I just appended it as a hint.

**Root cause:** `1.1.2` changed `useNextHeaders.js` from a top-level static import to a dynamic `require()` inside the function body:

```js
// 1.1.2 — BROKEN on Lambda
var useNextHeaders = function useNextHeaders() {
    var isServer = typeof window === 'undefined';
    if (isServer) {
        // Dynamic require to avoid bundling next/headers in client components
        var nextHeadersModule = require('next/headers'); // 💥 fails at runtime
        ...
    }
};
```

```
// 1.0.7 — works correctly
var _headers = require("next/headers"); // static, top-level — bundler resolves this correctly
```

The dynamic `require()` bypasses Next.js's static analysis at build time. The `next/headers` module is never bundled into the Lambda output, so it is absent at runtime.

1.0.7's static top-level require is correctly analyzed by the bundler and included in the Lambda bundle.

### 🔍 Steps to reproduce

1. Use Next.js App Router on Vercel (or any Node.js Lambda/serverless environment)
2. Use `<InstantSearchNext>` from `react-instantsearch-nextjs`
3. Upgrade `react-instantsearch-nextjs` from `1.0.7` to `1.1.2`
4. Deploy — build succeeds, but all pages rendering `<InstantSearchNext>` return 500 at runtime

### Live reproduction

N/A — only reproducible in a serverless/Lambda deployment (not next dev or next start on a regular server, where `next/headers` is available in the Node.js module graph)

### 💭 Expected behavior

`react-instantsearch-nextjs@1.1.2` should work on Vercel / serverless deployments the same as `1.0.7`.

The dynamic `require('next/headers')` pattern does not work in Lambda environments where the module must be statically bundled at build time. The intent (avoiding bundling `next/headers` in client components) could likely be achieved via a different mechanism — for example, marking the file with a server-only guard, using Next.js `serverExternalPackages` or splitting the server/client entry points in `package.json` `exports`.

### Package version

react-instantsearch-nextjs: 1.1.2 (regression from 1.0.7) react-instantsearch: 7.28.0 next: 16.1.6

### Operating system

All (only affects serverless/Lambda deployment targets)

### Browser

All

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/packages/react-instantsearch-nextjs/src/useNextHeaders.tsx
+++ b/packages/react-instantsearch-nextjs/src/useNextHeaders.tsx
@@ -1,3 +1,4 @@
+import { headers } from 'next/headers';
 import { use } from 'react';
 
 type NextHeaders = {
@@ -26,11 +27,7 @@ export const useNextHeaders = (): Headers => {
   const isServer = typeof window === 'undefined';
 
   if (isServer) {
-    // Dynamic require to avoid bundling next/headers in client components
-    const nextHeadersModule = require('next/headers') as {
-      headers: () => HeadersResult;
-    };
-    const nextHeaders = nextHeadersModule.headers();
+    const nextHeaders = headers() as HeadersResult;
     return isPromise(nextHeaders) ? use(nextHeaders) : nextHeaders;
   }
 

